model/api/src/main/java/org/jboss/aerogear/unifiedpush/api/WebPushVariant.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.aerogear.unifiedpush.api;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import nl.martijndwars.webpush.Utils;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.Security;
+
+/**
+ * The WebPush variant uses self-generated (and optionally provided) VAPID keys for push.
+ */
+public class WebPushVariant extends Variant {
+
+    private static final long serialVersionUID = -1873585264296190331L;
+
+    static {
+        if (Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
+            Security.addProvider(new BouncyCastleProvider());
```

Comment:
This won't be allowed in many secured environment (usually the security manager is configured to disallow this).
I would prefer to pass the provider directly to the functions that needs to use it ( I guess in this case they are `Utils.loadPublicKey` and `Utils.loadPrivateKey`